vortex-duckdb/src/convert/dtype.rs

```diff
@@ -119,24 +119,28 @@ impl LogicalType {
         let duckdb_type = match temporal_metadata {
             TemporalMetadata::Date(TimeUnit::D) => DUCKDB_TYPE::DUCKDB_TYPE_DATE,
             TemporalMetadata::Date(time_unit) => {
-                return Err(vortex_err!("Invalid TimeUnit {} for date", time_unit));
+                vortex_bail!("Invalid TimeUnit {} for date", time_unit);
             }
             TemporalMetadata::Time(TimeUnit::Us) => DUCKDB_TYPE::DUCKDB_TYPE_TIME,
             TemporalMetadata::Time(time_unit) => {
-                return Err(vortex_err!("Invalid TimeUnit {} for time", time_unit));
+                vortex_bail!("Invalid TimeUnit {} for time", time_unit);
             }
-            TemporalMetadata::Timestamp(time_unit, tz) => {
-                if tz.is_some() {
-                    return Err(vortex_err!("Timestamp with timezone is not yet supported"));
+            TemporalMetadata::Timestamp(time_unit, tz) => match time_unit {
+                TimeUnit::Ns => DUCKDB_TYPE::DUCKDB_TYPE_TIMESTAMP_NS,
+                TimeUnit::Us => {
+                    if let Some(tz) = tz {
+                        if tz != "UTC" {
+                            vortex_bail!("Invalid timezone for timestamp: {tz}");
+                        }
```

Comment:
Hmmm, this needs some check against duckdb

Reply:
We could support other timezones here by running conversion on the data before export but that can be a followup